client/packages/invoices/src/InboundShipment/api/hooks/document/useListInternalOrders.ts

```diff
@@ -0,0 +1,18 @@
+import { useQuery } from '@openmsupply-client/common';
+import { useInboundApi } from '../utils/useInboundApi';
+
+const MILLISECONDS_PER_MINUTE = 60 * 1000;
+const POLLING_INTERVAL_MS = 3 * MILLISECONDS_PER_MINUTE;
+const STALE_TIME_MS = 1 * MILLISECONDS_PER_MINUTE;
+
+export const useListInternalOrders = (otherPartyId: string) => {
+  const api = useInboundApi();
+
+  return {
+    ...useQuery(
+      api.keys.listInternalOrders(otherPartyId),
+      () => api.get.listInternalOrders(otherPartyId),
+      { cacheTime: POLLING_INTERVAL_MS, staleTime: STALE_TIME_MS }
```

Comment:
This is cool - I haven't seen this used before. Is this for constant refreshing for the list in the modal?

Reply:
It doesn't constantly refresh. It just lowers the cache/stale time so that the api does get refetched if called

Reply:
Nice thanks Roxy - I was having a little read on tanStack also